app/lib/actions/katello/repository/update_redhat_repository.rb

```diff
@@ -0,0 +1,30 @@
+module Actions
+  module Katello
+    module Repository
+      class UpdateRedhatRepository < Actions::EntryAction
+        def plan(repo)
+          action_subject repo
+          repo.root.update_attributes!(:url => upstream_url(repo)) if repo.library_instance?
```

Comment:
If we pass 5 repos from a single content object we'll be updating the root repo 5 times - probably worth avoiding that

Reply:
```repo.library_instance?``` will ensure this  only runs once :)